File: Examples/Node.js/Document/AppendPdfFiles.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var PdfApi = require('asposepdfcloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../config/config.json');
var data_path = '../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Pdf API SDK
var pdfApi = new PdfApi(config);

// Set input file name
var name = "Sample.pdf";
var startPage = 2;
var endPage = 3;
var appendFileName = "sample-input.pdf";
var appendDocumentBody = {
		'Document' : appendFileName,
		'StartPage' : startPage,
		'EndPage' : endPage
};

try {
// Upload source file to aspose cloud storage
storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
	assert.equal(responseMessage.status, 'OK');

	storageApi.PutCreate(appendFileName, null, null, data_path + appendFileName , function(responseMessage) {
		assert.equal(responseMessage.status, 'OK');

		// Invoke Aspose.Pdf Cloud SDK API to append multiple PDF files
		pdfApi.PostAppendDocument(name, null, startPage, endPage, null, null, appendDocumentBody, function(responseMessage) {
				assert.equal(responseMessage.status, 'OK');

				// Download appended document from storage server
				storageApi.GetDownload(name, null, null, function(responseMessage) {
					assert.equal(responseMessage.status, 'OK');
					var outfilename = 'appended_out' + name;
					var writeStream = fs.createWriteStream(data_path  + outfilename);
					writeStream.write(responseMessage.body);
					});
			});
		});
	});

}
catch (e) 
{
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Python/Examples/AppendPDFFiles.py
import asposepdfcloud
from asposepdfcloud.PdfApi import PdfApi
from asposepdfcloud.PdfApi import ApiException
from asposepdfcloud.models import Signature
from asposepdfcloud.models import Rectangle

import asposestoragecloud
from asposestoragecloud.StorageApi import StorageApi
from asposestoragecloud.StorageApi import ResponseMessage

apiKey = "XXXXX" #sepcify App Key
appSid = "XXXXX" #sepcify App SID
apiServer = "http://api.aspose.com/v1.1"
data_folder = "../../data/"

#Instantiate Aspose Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, True)
storageApi = StorageApi(storage_apiClient)
#Instantiate Aspose Pdf API SDK
api_client = asposepdfcloud.ApiClient.ApiClient(apiKey, appSid, True)
pdfApi = PdfApi(api_client);

#set input file name
name = "sample-input.pdf"
signatureFileName = "pkc7-sample.pfx"
pageNumber = 1

body = Signature.Signature()
body.Authority =  "Farooq Sheikh"
body.Location = "Rawalpindi"
body.Contact =  "farooq.sheikh@aspose.com"
body.Date =  "06/24/2015 2:00:00.000 AM"
body.FormFieldName = "Signature1"
body.Password = "aspose"

rect = Rectangle.Rectangle();
rect.X = 100
rect.Y = 100
rect.Height = 100
rect.Width = 200
body.Rectangle = rect
body.SignaturePath = signatureFileName
body.SignatureType = "PKCS7"
body.Visible = True


try:
    #upload file to aspose cloud storage
    response = storageApi.PutCreate(name, data_folder + name)
    response = storageApi.PutCreate(signatureFileName, data_folder + signatureFileName)

    #invoke Aspose.Pdf Cloud SDK API to sign PDF document
    response = pdfApi.PostSignPage(name, pageNumber, body)


    if response.Status == "OK":
        #download signed pdf from response
        response = storageApi.GetDownload(Path=name)
        outfilename = "c:/temp/" + name
        with open(outfilename, 'wb') as f:
                    for chunk in response.InputStream:
                        f.write(chunk)

except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message

